### ISSUE TITLE
Pupil Mobile - recording onto Moto Z2 Play's SD card does not have info.csv file

### ISSUE THREAD

**Issue by Fin-13** [pupil-labs/pupil-mobile-app#10]
i was able to open pupil mobile recordings in pupil player however the menus were not visible.  i could click through the clickable buttons which would highlight in turn however their separate menus would not display.
i attempted to reinstall the pupil player app and following reinstallation, the recordings would no longer load at all.  attempting to load the recordings would result in an error message indicating that the file was not a valid pupil recording.
it appears the info.csv files went missing and once a sample info.csv file is copied into the recording files they can now be viewed again.

**Comment by willpatera** [comment]
@saki4510t It appears that the `info.csv` file was not created for @Fin-13 on the latest version of the Pupil Mobile app. The user recorded directly to the SD card of a Moto Z2 Play device. Can you re-create this issue?

**Comment by saki4510t** [comment]
@willpatera I can't re-produce this issue yet, it may depends on the condition of SD card. I will try to find potential causes on app.

**Comment by AprilRG** [comment]
I had a similar issue with the info.csv file not being created. The recording was randomly shutting off early. I thought it was a storage issue, but I added a 256 GB micro SD to my phone and set it to save to the SD card. But it seems to stop recording randomly still. So this happened with and without the SD card. When the problem occurs, the info file is missing from the folder.  It doesn't always cut the recording early either.

**Comment by papr** [comment]
@saki4510t This is a reproducible issue. @AprilRG found out that the recording is terminated if the wifi disconnects.

Steps to reproduce:
1. Connect to a wifi
2. Open Pupil Mobile and start a local recording
3. Turn off wifi via Quick Settings

Behavior on One Plus 3 with Pupil Mobile 0.21.2: UI closes. Opening app again shows that the recording is not running anymore. Inspecting the recording folder within the folder view of the app shows that info.csv was not stored.

**Comment by saki4510t** [comment]
@papr 
Turning off Wi-Fi (by air plane mode, by setting etc.) is almost as same as removing network card from usual PC while networking.  
Unfortunately zmq (network libraries that Pupil app uese) can never handle these type events well. And zmq many times raises un-recorveralble exception and crashes before any handling on app / before trying stopping zmq when app receives network state change event (even if app has signal handler).
The exception will occur at anywhere in zmq (libzmq, czmq, cppzmq and zyre) and hard to improve. So there are no effective improvement than avoiding such operation now.

**Comment by AprilRG** [comment]
I had this occur again to 2 of 5 recordings I made over the weekend. I was not connected to wifi the entire time and I did not see any error messages. I have player v1.6.11 and the most recent release for the pupil mobile app as well. This time loading a sample info.csv file did not allow it to load in player. The error in the command line reads "launchables.player: Could not generate world timestamps from eye timestamps. This is an invalid recording."
